File: app/src/main/java/name/mikanoshi/customiuizer/tasker/UnlockSettings.java
package name.mikanoshi.customiuizer.tasker;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.RadioGroup;

import androidx.appcompat.app.AppCompatActivity;

import name.mikanoshi.customiuizer.R;

public class UnlockSettings extends AppCompatActivity {

	@Override
	protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);

		setContentView(R.layout.tasker_unlock);

		Button ok = findViewById(R.id.force_ok);
		ok.setOnClickListener(new View.OnClickListener() {
			@Override
			public void onClick(View v) {
				int opt = ((RadioGroup)findViewById(R.id.force_option)).getCheckedRadioButtonId();
				int val = -1;
				if (opt == R.id.force_locked) val = 0;
				else if (opt == R.id.force_unlocked) val = 1;
				Intent resultIntent = new Intent();
				resultIntent.putExtra(Constants.EXTRA_STRING_BLURB, getString(val == 1 ? R.string.system_noscreenlock_force_unlocked : (val == 0 ? R.string.system_noscreenlock_force_locked : R.string.system_noscreenlock_force_off)));
				Bundle bundle = new Bundle();
				bundle.putInt("system_noscreenlock_force", val);
				//bundle.putString("net.dinglisch.android.tasker.extras.VARIABLE_REPLACE_KEYS", "lockstate");
				//resultIntent.putExtra("net.dinglisch.android.tasker.RELEVANT_VARIABLES", new String[] { "%lockstate\nLock state\nCurrent state of lock override:<br><b>0</b> force locked<br><b>1</b> force unlocked<br><b>-1</b> do not force"});
				//resultIntent.putExtra("net.dinglisch.android.tasker.extras.REQUESTED_TIMEOUT", 3000);
				resultIntent.putExtra(Constants.EXTRA_BUNDLE, bundle);
				setResult(Activity.RESULT_OK, resultIntent);
				finish();
			}
		});

		Bundle bundle = getIntent().getBundleExtra(Constants.EXTRA_BUNDLE);
		if (bundle != null) {
			int opt = bundle.getInt("system_noscreenlock_force", -1);
			((RadioGroup)findViewById(R.id.force_option)).check(opt == 0 ? R.id.force_locked : (opt == 1 ? R.id.force_unlocked : R.id.force_off));
		}
	}

}

File: app/src/main/java/tv/withaibuild/customiuizer/tasker/UnlockSettings.kt
package tv.withaibuild.customiuizer.tasker

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.widget.Button
import android.widget.RadioGroup
import androidx.appcompat.app.AppCompatActivity
import tv.withaibuild.customiuizer.R

class UnlockSettings : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContentView(R.layout.tasker_unlock)

        val ok = findViewById<Button>(R.id.force_ok)
        ok.setOnClickListener {
            val opt = findViewById<RadioGroup>(R.id.force_option).checkedRadioButtonId
            val lockState = when (opt) {
                R.id.force_locked -> 0
                R.id.force_unlocked -> 1
                else -> -1
            }

            val stringRes = when (lockState) {
                1 -> R.string.system_noscreenlock_force_unlocked
                0 -> R.string.system_noscreenlock_force_locked
                else -> R.string.system_noscreenlock_force_off
            }

            val resultIntent = Intent().apply {
                putExtra(Constants.EXTRA_STRING_BLURB, getString(stringRes))
                val bundle = Bundle().apply {
                    putInt("system_noscreenlock_force", lockState)
                }
                putExtra(Constants.EXTRA_BUNDLE, bundle)
            }
            setResult(Activity.RESULT_OK, resultIntent)
            finish()
        }

        val bundle = intent.getBundleExtra(Constants.EXTRA_BUNDLE)
        if (bundle != null) {
            val opt = bundle.getInt("system_noscreenlock_force", -1)
            val checkedId = when (opt) {
                0 -> R.id.force_locked
                1 -> R.id.force_unlocked
                else -> R.id.force_off
            }
            findViewById<RadioGroup>(R.id.force_option).check(checkedId)
        }
    }
}
